From 1efb77a14a14463de9cfe6188675b47ab9a22cc2 Mon Sep 17 00:00:00 2001
From: Matthias Runge <mrunge@redhat.com>
Date: Tue, 11 Nov 2014 14:47:47 +0100
Subject: [PATCH] Fix dashboard nav, when panelgroup empty

The identity dashboard breaks with a common pattern
and doesn't use a panel group to group panels. When not
using accordion navigation, the last level of navigation is
not shown at all.

Closes-bug: #
---
 horizon/templatetags/horizon.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/horizon/templatetags/horizon.py b/horizon/templatetags/horizon.py
index f798b4a..5752775 100644
--- a/horizon/templatetags/horizon.py
+++ b/horizon/templatetags/horizon.py
@@ -23,6 +23,7 @@ from django.utils.encoding import force_text
 from django.utils import translation
 from django.utils.translation import ugettext_lazy as _
 
+import horizon
 from horizon.base import Horizon  # noqa
 from horizon import conf
 
@@ -116,7 +117,10 @@ def horizon_dashboard_nav(context):
                     panel.can_access(context)):
                 allowed_panels.append(panel)
         if allowed_panels:
-            non_empty_groups.append((group.name, allowed_panels))
+            if group.name==None:
+                non_empty_groups.append((horizon.Dashboard(),allowed_panels))
+            else:
+                non_empty_groups.append((group.name, allowed_panels))
 
     return {'components': SortedDict(non_empty_groups),
             'user': context['request'].user,
